root/doc/DocPage.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import Layout from '../layout';
+
+import DocSearchBox from './components/DocSearchBox';
+
+type DocPageT = {|
+  +content: string,
+  +hierarchy: $ReadOnlyArray<string>,
+  +title: string,
+  +version: number,
+|};
+
+type Props = {|
+  +id: string,
+  +page: DocPageT,
+  +useGoogleCustomSearch: boolean,
+  +wikiServer: string,
+|};
+
+const DocPage = ({
+  id,
+  page,
+  useGoogleCustomSearch,
+  wikiServer,
+}: Props) => {
+  let doc = '';
+  let lastDoc = '';
+  const wikiPageUrl = `//${wikiServer}/${id}`;
+  const wikiPage = (
+    <a
+      className="internal"
+      href={wikiPageUrl}
+    >
+      {page.title}
+    </a>
+  );
+  const wikiVersion = (
+    <a
+      className="internal"
+      href={`//${wikiPageUrl}?oldid=${page.version}`}
+    >
+      {'#' + page.version}
+    </a>
+  );
+  return (
+    <Layout fullWidth no_icons title={page.title}>
+      <div className="wikicontent" id="content">
+        {useGoogleCustomSearch ? <DocSearchBox /> : null}
+
+        <h1 className="hierarchy-links">
+          {page.hierarchy.map((link, index, array) => {
+            lastDoc = doc;
+            doc = index === 0 ? link : lastDoc + '/' + link;
+            const isLast = array.length - 1 === index;
+            return (
+              <React.Fragment key={index + '-' + link}>
+                <a
+                  href={'/doc/' + doc.replace(/ /g, '_')}
+                  style={{fontWeight: isLast ? 'bold' : ''}}
+                >
+                  {link}
+                </a>
+                {isLast ? null : ' / '}
+              </React.Fragment>
+            );
+          })}
+        </h1>
+
+        {page.version ? null : (
+          <div className="wikidocs-header">
+            {exp.l(`This page has not been reviewed by our documentation team
+                    ({more_info|more info}).`,
+                   {more_info: '/doc/WikiDocs'})}
+          </div>
+        )}
+
+        <div dangerouslySetInnerHTML={{__html: page.content}} />
```

Comment:
This div didn't exist on TT, but is needed in React - shouldn't be a problem hopefully.